PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -9697,6 +10047,68 @@ void DetermineOccupancy(T const& collision)
 
 //============================================================
 
+template <eRecSim rs, typename T1, typename T2>
+void DetermineInteractionRate(T1 const& collision, T2 const&)
+{
+  // Determine interaction rate.
+
+  // a) Determine interaction rate only for eRec;
+  // b) For all other cases, set interaction rate to -1 for the time being;
+  // c) Print interaction rate and run duration for the audience...
+
+  if (tc.fVerbose) {
+    StartFunction(__FUNCTION__);
+  }
+
+  // a) Determine interaction rate only for eRec:
+  if constexpr (rs == eRec) {
+    auto bc = collision.template foundBC_as<T2>(); // TBI 20241126 this seems to works as expected, but check and validate further
+                                                   // cout << bc.runNumber() << endl; // works as expected
+                                                   // cout << bc.timestamp() << endl;  // works as expected
+                                                   // TBI 20241126 I have the same code snippet at 2 other locations, I shall unify at some point
+
+    // Get start timestamp and end timemstamp for this run in miliseconds, and convert both of them in seconds:
+    // o see O2/CCDB/src/BasicCCDBManager.cxx, O2/CCDB/include/CCDB/BasicCCDBManager.h
+    // o example usage in O2Physics/PWGLF/TableProducer/Common/zdcSP.cxx
+    auto runDuration = ccdb->getRunDuration(bc.runNumber()); // this is total run duration, not the current one (see below)
+    int64_t startOfRun = std::floor(runDuration.first * 0.001);
+    int64_t endOfRun = std::ceil(runDuration.second * 0.001);
+    int64_t runDurationInSec = endOfRun - startOfRun; // yes, this is now in seconds
```

Comment:
Consider doing this at run level and not at collision level. They will not change, right?

Reply:
Yes, well-spotted, thanks!